lib/src/controls/pickers/calendar_view.dart

```diff
@@ -0,0 +1,1565 @@
+
+  /// Only a single date can be selected.
+  single,
+
+  /// A range of dates can be selected.
+  range,
+
+  /// Multiple, non-contiguous dates can be selected.
+  multiple,
+}
+
+/// Defines the display modes for the [CalendarView].
+enum CalendarViewDisplayMode {
+  /// Displays the calendar in month view.
+  month,
+
+  /// Displays the calendar in year view.
+  year,
+
+  /// Displays the calendar in decade view.
+  decade,
+}
+
+/// Represents the selection data for the [CalendarView].
+class CalendarSelectionData {
+  /// The list of selected dates.
+  final List<DateTime> selectedDates;
+
+  /// The start date of the selection range, if applicable.
+  final DateTime? startDate;
+
+  /// The end date of the selection range, if applicable.
+  final DateTime? endDate;
+
+  CalendarSelectionData({
+    required this.selectedDates,
+    this.startDate,
+    this.endDate,
+  }) : assert(
+         startDate == null || endDate == null || startDate.isBefore(endDate),
+         'Start date must be before end date if both are provided.',
+       );
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) return true;
+
+    return other is CalendarSelectionData &&
+        listEquals(other.selectedDates, selectedDates) &&
+        other.startDate == startDate &&
+        other.endDate == endDate;
+  }
+
+  @override
+  int get hashCode =>
+      selectedDates.hashCode ^ startDate.hashCode ^ endDate.hashCode;
+}
+
+/// A calendar view lets a user view and interact with a calendar that they can
+/// navigate by month, year, or decade. A user can select a single date or a
+/// range of dates. It doesn't have a picker surface and the calendar is always
+/// visible.
+///
+/// ![](https://learn.microsoft.com/en-us/windows/apps/design/controls/images/calendar-view-3-views.png)
+///
+/// See also:
+///
+///  * [DatePicker], which provides a more compact date selection interface.
+///  * [TimePicker], which allows users to select a time.
+///  * <https://learn.microsoft.com/en-us/windows/apps/design/controls/calendar-view>
+class CalendarView extends StatefulWidget {
+  /// The initial start date for selection.
+  final DateTime? initialStart;
+
+  /// The initial end date for range selection.
+  final DateTime? initialEnd;
+
+  /// Callback invoked when the selection changes.
+  ///
+  /// This provides a [CalendarSelectionData] object containing the selected
+  /// dates, start date, and end date.
+  ///
+  /// See also:
+  ///
+  ///   * [CalendarSelectionData] for details on the selection data structure.
+  final ValueChanged<CalendarSelectionData>? onSelectionChanged;
+
+  /// The minimum date that can be selected.
+  final DateTime? minDate;
+
+  /// The maximum date that can be selected.
+  final DateTime? maxDate;
+
+  /// The last date in the date range that is available in the calendar..
+  final DateTime? displayDateEnd;
+
+  /// The first date that is available in the calendar.
+  final DateTime? displayDateStart;
```

Comment:
Documentation could be improved saying the difference between these properties. We could also create a separate object called `DateBounds`(?) and reduce these properties.